setuper/additional_projects.py

```diff
@@ -0,0 +1,117 @@
+            "feature_flags": ["REQUIRE_AUTHENTICATION", "TAKE_GPS_ON_FORM"],
+            "linked_forms": ["Equipment/Pop/Social mob./Microplans"],
+        },
+        {
+            "name": "Children vaccination/Vaccination des enfants",
+            "app_id": f"{account_name}.children",
+            "feature_flags": ["REQUIRE_AUTHENTICATION", "ENTITY"],
+            "linked_forms": ["Child - Registration", "Child - Follow up"],
+        },
+    ]
+    return projects
+
+
+def feature_flags_mapper(project, iaso_client):
+    existing_feature_flags = iaso_client.get("/api/featureflags/")
+    feature_flags = []
+    for feature_flag_code in project["feature_flags"]:
+        project_feature_flag = [
+            feature_flag
+            for feature_flag in existing_feature_flags["featureflags"]
+            if feature_flag["code"] == feature_flag_code
+        ]
+        if len(project_feature_flag) > 0:
+            feature_flags.append(project_feature_flag[0])
+    return feature_flags
+
+
+def get_project_ids(created_or_updated_projects, iaso_client):
+    existing_projects = iaso_client.get("/api/projects/")["projects"]
+    project_ids = []
+    for current_project in created_or_updated_projects:
+        project_id = [project for project in existing_projects if project["app_id"] == current_project["id"]]
+        project_ids.append(project_id[0]["id"])
+    return project_ids
+
+
+def update_org_unit_types_with_new_projects(iaso_client, project_ids):
+    org_unit_types = iaso_client.get("/api/v2/orgunittypes/")["orgUnitTypes"]
+    updated_types = []
+    for org_unit_type in org_unit_types:
+        linked_project_ids = [project["id"] for project in org_unit_type["projects"]]
+        org_unit_type["project_ids"] = linked_project_ids
+        org_unit_type["project_ids"].extend(project_ids)
+        type = iaso_client.patch(f"/api/v2/orgunittypes/{org_unit_type['id']}/", json=org_unit_type)
+        updated_types.append(type)
+    return updated_types
+
+
+def link_created_projects_to_main_data_source(account_name, iaso_client, project_ids):
+    data_sources = iaso_client.get("/api/datasources/")["sources"]
+    data_source = [data_source for data_source in data_sources if data_source["name"] == account_name]
+
+    if len(data_source) > 0:
+        default_project = data_source[0]["projects"][0][0]["id"]
+        project_ids.append(default_project)
+        data_source[0]["project_ids"] = project_ids
+        data_source[0]["default_version_id"] = data_source[0]["default_version"]["id"]
+        iaso_client.put(f"/api/datasources/{data_source[0]['id']}/", json=data_source[0])
+
+
+def create_projects(account_name, iaso_client):
+    print(f"-- Creating 3 additional projects for account :{account_name}")
+    projects = projects_mapper(account_name)
+    created_projects = []
+    for project in projects:
+        project["linked_forms"] = None
+        project["feature_flags"] = feature_flags_mapper(project, iaso_client)
```

Comment:
```suggestion

    flags = iaso_client.get("/api/featureflags/")
    existing_flags = set([flag["code"] for flag in flags["featureflags"]])

    for project in projects:
        project["linked_forms"] = None
        project["feature_flags"] = set(project["feature_flags"]).intersection(existing_flags)
```

We could save some HTTP queries by avoiding calling `feature_flags_mapper() -> GET /api/featureflags/` in each iteration of the loop.

What do you think?

Reply:
Nice idea. But just a small change as the  payload for `post: /api/apps/` need the whole object for project features flags , I would suggest:
-  to remove line 78 and keep all existing flags 
- on line 82, to filter on existing flags to keep only flags defined on the current project : `project["feature_flags"] = [flag for flag in flags["featureflags"] if flag["code"] in project["feature_flags"]]`

Reply:
Yup, this works!